### PR TITLE
DC-2844: fixing and refactoring IT tests

### PR DESCRIPTION
**Pull request by rwalpole**

### CODE REVIEW

File: it/PostCustomerReadTimeISpec.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import play.api.http.{ ContentTypes, HeaderNames }
+
+@DoNotDiscover
+@SuppressWarnings(Array("org.wartremover.warts.All"))
+class PostCustomerReadTimeISpec extends ISpec {
+
+  "A POST request to /secure-messaging/conversation/{client}/{conversationId}/read-time" should {
+
+    "return CREATED when read time added to DB" in {
+      createConversation
```
Comment (alexflav23) [inline it/PostCustomerReadTimeISpec.scala:30]: This is the only dodgy looking thing. Nullary methods that cause mutation should use `()` by convention in Scala. Can this not be a chain future or something, so the action can be controlled? Not super important, can merge without ultimately.

File: it/PostCustomerReadTimeISpec.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        wsClient
+          .url(resource("/secure-messaging/conversation/cdcm/D-80542-20201120/read-time"))
+          .withHttpHeaders(buildEoriToken(VALID_EORI), (HeaderNames.CONTENT_TYPE, ContentTypes.JSON))
+          .post(new File("./it/resources/read-time.json"))
+          .futureValue
+      response.status mustBe CREATED
+    }
+
+    "return BAD REQUEST when read time not added to DB" in {
+      createConversation
```
Comment (alexflav23) [inline it/PostCustomerReadTimeISpec.scala:41]: Same as above

**Review by alexflav23 — DISMISSED** [review]

File: it/PostCustomerReadTimeISpec.scala (hunk 1 above)
Comment (rwalpole) [reply it/PostCustomerReadTimeISpec.scala]: Oh, I think you have put your finger on one of the problems I was having with these tests (intermittent failure) @alexflav23 Have updated. Thank you :)

File: it/PostCustomerReadTimeISpec.scala (hunk 2 above)
Comment (rwalpole) [reply it/PostCustomerReadTimeISpec.scala]: See comment above.

**Review by jim-collins — APPROVED** [review]

**Pull request merged**